classes/security/authorization/internal/ApiCsrfMiddleware.inc.php

```diff
@@ -34,9 +34,21 @@ public function __construct(APIHandler $handler) {
 	 * @param SlimRequest $slimRequest request
 	 * @param SlimResponse $response response
 	 * @param callable $next Next middleware
+	 *
 	 * @return SlimResponse
 	 */
-	public function __invoke($slimRequest, $response, $next) {
+	public function __invoke(
+		$slimRequest,
+		$response,
+		$next
+	) {
+		if (!$this->tokenFilled()) {
+			return $response->withJson([
+				'error' => 'api.api_token.should.be.filled',
+				'errorMessage' => __('api.api_token.should.be.filled'),
+			], 403);
+		}
```

Comment:
This is causing errors for me when I use the editorial backend. That's because the editorial backend UI (like the submissions page) uses the API without an API Token. Instead it uses session cookies to authenticate the user. The API Token is only needed when making requests to the API from a third-party domain, where the session cookies are not available.

I think you can revert all of the changes in `ApiCsrfMiddleware.inc.php`. What you've done in `ApiTokenDecodingMiddleware.inc.php` seems to do the trick.